transport/TransportImpl.cpp

```diff
@@ -2328,4 +2333,27 @@ void TransportImpl::setTag(const char* tag)
     utils::strncpy(_tag, tag, sizeof(_tag));
 }
 
+void TransportImpl::onIceDiscardCandidate(ice::IceSession* session,
+    ice::IceEndpoint* endpoint,
+    const transport::SocketAddress& sourcePort)
+{
+    for (auto& udpEndpoint : _rtpEndpoints)
```

Comment:
I am wondering if this lookup is really necessary. What bad could happen if you call the `dpEndpoint->unregisterListener(sourcePort, this);` without check if it exists in `_rtpEndpoints` or ìn `_rtcpEndpoints`

Reply:
I would still have to go through all endpoints and call unregister, then the endpoint would ignore it because it cannot find the sourceport, perhaps.